### ISSUE TITLE
Flexvol deprecation

### ISSUE THREAD

**Issue by uristernik** [projectcalico/calico#7600]
FlexVolumes are [deprecated as of Kubernetes v1.23](https://kubernetes.io/blog/2021/12/07/kubernetes-1-23-release-announcement/#deprecation-of-flexvolume). I am currently using calico v3.23.5 with flexvol v3.24.3.
I saw this version was tested in v1.23 [link](https://docs.tigera.io/archive/v3.23/getting-started/kubernetes/requirements#:~:text=Kubernetes%20requirements-,Supported%20versions,-We%20test%20Calico)
## Expected Behavior

Use CSI driver

## Current Behavior

Flexvol as init container

## Context

K8s upgrade.
BTW the flexvol is used to import libs from the underlying host os?

## Your Environment

* Calico version: calico v3.23.5, flexvol v3.24.3
* Orchestrator version (e.g. kubernetes, mesos, rkt): v1.22
* EKS cluster

**Comment by mgleung** [comment]
Hi @uristernik . You're right that flexvols are deprecated in Kubernetes 1.23. We added CSI support in v3.23.0 so with your version of Calico, you should be running a version that has CSI set up instead of flexvol. How are you installing v3.23.5?

**Comment by caseydavenport** [comment]
> BTW the flexvol is used to import libs from the underlying host os?

Nope, it's just used for setting up a socket for communication between our host-based policy engine (Felix) and our L7 policy sidecar (Dikastes)

**Comment by uristernik** [comment]
Hey thanks for the replay! @mgleung 
I am installing it using helm (flux helm controller). I am actually upgrading to v3.25.1 right now and wondered:
1. If flexvol isn't used anymore why the pod still has init container with flexvol image?
2. Is there a way to use an existing CSI driver daemonset instead of installing another daemonset?
